src/emulator/gui/src/profiles_manager_dialog.cpp

```diff
@@ -0,0 +1,189 @@
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License along
+// with this program; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+#include "private.h"
+
+#include <config/functions.h>
+#include <host/state.h>
+#include <misc/cpp/imgui_stdlib.h>
+#include <util/log.h>
+
+namespace gui {
+
+void draw_profiles_manager_dialog(GuiState &gui, HostState &host) {
+    ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
+    ImGui::Begin("Profiles Manager", &gui.configuration_menu.profiles_manager_dialog, ImGuiWindowFlags_AlwaysAutoResize);
+    ImGui::PopStyleColor();
+    auto update_online_id = std::find(host.cfg.online_id.begin(), host.cfg.online_id.end(), gui.online_id);
+
+    // Online ID
+    ImGui::InputTextWithHint("##online_id", "Write your online ID", &gui.online_id);
+    if (ImGui::IsItemHovered())
+        ImGui::SetTooltip("Set your Online ID with maximum 16 character.");
+    ImGui::SameLine();
+    if (ImGui::Button("Add Online ID") && !gui.online_id.empty())
+        ImGui::OpenPopup("Online ID");
+    ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
+    if (ImGui::BeginPopupModal("Online ID", NULL, ImGuiWindowFlags_AlwaysAutoResize)) {
+        ImGui::PopStyleColor();
+        if (gui.online_id.length() > 16)
+            ImGui::Text("Error: Need reduce longer for online ID.\nActuall longer: %d.", gui.online_id.length());
+        else if (update_online_id != host.cfg.online_id.end())
+            ImGui::Text("Error: '%s' online ID already exist.", gui.online_id.c_str());
+        else
+            ImGui::Text("Succes: '%s' online ID added.", gui.online_id.c_str());
+
+        ImGui::Separator();
+        if (ImGui::Button("OK", ImVec2(128, 0))) {
+            if (update_online_id == host.cfg.online_id.end() && !(gui.online_id.length() > 16)) {
+                host.cfg.online_id.push_back(gui.online_id);
+                host.cfg.user_id += (int)host.cfg.online_id.size() - host.cfg.user_id;
+                gui.online_id.clear();
+            }
+            ImGui::CloseCurrentPopup();
+        }
+        ImGui::EndPopup();
+    } else
+        ImGui::PopStyleColor();
+
+    ImGui::PushItemWidth(300);
+    if (ImGui::ListBoxHeader("##online_id_list", (int)host.cfg.online_id.size(), 3)) {
+        for (auto i = 0; i < host.cfg.online_id.size(); i++) {
+            ImGui::Text("User ID: %02d, Online ID: %s", i, host.cfg.online_id[i].c_str());
+        }
+        ImGui::ListBoxFooter();
+    }
+    ImGui::PopItemWidth();
+    ImGui::Spacing();
+    if (host.cfg.online_id.size() > 1) {
+        ImGui::Combo("Select your User", &host.cfg.user_id,
```

Comment:
No capital "U" **or** capitalize the "Y."

Reply:
Good see ^^